pkg/proxy/iptables/proxier.go

```diff
@@ -1063,41 +1063,53 @@ func (proxier *Proxier) syncProxyRules() {
 		// worthwhile to make a new per-service chain for nodeport rules, but
 		// with just 2 rules it ends up being a waste and a cognitive burden.
 		if svcInfo.nodePort != 0 {
-			// Hold the local port open so no other process can open it
-			// (because the socket might open but it would never work).
-			lp := localPort{
-				desc:     "nodePort for " + svcName.String(),
-				ip:       "",
-				port:     svcInfo.nodePort,
-				protocol: protocol,
-			}
-			if proxier.portsMap[lp] != nil {
-				glog.V(4).Infof("Port %s was open before and is still needed", lp.String())
-				replacementPortsMap[lp] = proxier.portsMap[lp]
-			} else {
-				socket, err := proxier.portMapper.OpenLocalPort(&lp)
-				if err != nil {
-					glog.Errorf("can't open %s, skipping this nodePort: %v", lp.String(), err)
-					continue
+			if len(proxier.endpointsMap[svcName]) != 0 {
+				// Hold the local port open so no other process can open it
+				// (because the socket might open but it would never work).
+				lp := localPort{
```

Comment:
Even if we don't have any endpoints, we should hold the port open.  We don't want someone coming along later thinking they can use it.